src/test/java/com/puppycrawl/tools/checkstyle/grammar/GeneratedJavaTokenTypesTest.java

```diff
@@ -711,7 +719,48 @@ public void testTokenNumbering() {
                         + " 'GeneratedJavaTokenTypesTest' and verified"
                         + " that their old numbering didn't change")
             .that(tokenCount)
-            .isEqualTo(225);
+            .isEqualTo(226);
+    }
+
+    /**
+     * This test was created to make sure that new tokens are added to the 'tokens'
+     * block in the lexer grammar. If a new token is not added at the end of the list,
+     * it will become "mixed in" with the unused tokens and cause
+     * Collections#lastIndexOfSubList to return a -1 and fail the test.
+     */
+    @Test
+    public void testTokenHasBeenAddedToTokensBlockInLexerGrammar() {
+        final VocabularyImpl vocabulary = (VocabularyImpl) JavaLanguageLexer.VOCABULARY;
+        final String[] nullableSymbolicNames = vocabulary.getSymbolicNames();
+        final List<String> allTokenNames = Arrays.stream(nullableSymbolicNames)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+
+        // Since the following tokens are not declared in the 'tokens' block,
+        // they will always appear last in the list of symbolic names provided
+        // by the vocabulary.
+        final List<String> unusedTokenNames = List.of(
+                // reserved keywords that are not part of the language
+                "LITERAL_CONST", "LITERAL_GOTO",
```

Comment:
I'm confused, why are these here if they are not part of the language? Goto is not used?

Reply:
`goto` and `const` are reserved keywords in the java langauge, but they are not used for historical reasons. I included them as keywords in the lexer for completeness in regards to the language spec, but of course they are never used.

Reply:
Ok, they are not in `TokenTypes`.